fastcrypto-cli/src/tlock.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::Parser;
+use fastcrypto::error::{FastCryptoError, FastCryptoResult};
+use fastcrypto::groups::{bls12381, GroupElement, HashToGroupElement, Pairing};
+use fastcrypto::hash::{Blake2b256, HashFunction, Sha256};
+use fastcrypto::serde_helpers::ToFromByteArray;
+use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+use std::io::{Error, ErrorKind};
+
+#[derive(Parser)]
+#[command(name = "tlock-cli")]
+#[command(about = "Basic tlock interface (for quicknet).", long_about = None)]
+enum Command {
+    Encrypt(EncryptArguments),
+    Decrypt(DecryptArguments),
+    Verify(VerifyArguments),
+}
+
+/// Encrypt a plaintext. Returns a ciphertext (hex).
+#[derive(Parser, Clone)]
+struct EncryptArguments {
+    /// drand round number
+    #[clap(short, long)]
+    round: u64,
+
+    /// Plaintext to encrypt (hex, 32 bytes)
+    #[clap(short, long)]
+    plaintext: String,
+}
+
+/// Decrypt a ciphertext. Returns a plaintext (hex).
+#[derive(Parser, Clone)]
+struct DecryptArguments {
+    /// Ciphertext to decrypt (hex)
+    #[clap(short, long)]
+    ciphertext: String,
+
+    /// drand signature for the required round (hex, 48 bytes)
+    #[clap(short, long)]
+    round_signature: String,
+}
+
+/// Verify a drand signature for a given round.
+#[derive(Parser, Clone)]
+struct VerifyArguments {
+    /// drand round number
+    #[clap(short, long)]
+    round: u64,
+
+    /// drand signature (hex, 48 bytes)
+    #[clap(short, long)]
+    signature: String,
+}
+
+/// An encryption of 32 bytes message following https://eprint.iacr.org/2023/189.pdf.
```

Comment:
How do we encrypt shorter and larger msgs? Do we need to enable AES decryption in Sui?

Reply:
Today you can use the 32 bytes as an HMAC key, and impl an encryption of any size using HMAC(key, i) xor m_i where m_i is the i-th block of the message

Reply:
(for shorter messages, pad it if needed)